transaction/std/src/memo_builder/rth_memo_builder.rs

```diff
@@ -143,16 +148,25 @@ impl MemoBuilder for RTHMemoBuilder {
     /// Build a memo for a normal output (to another party).
     fn make_memo_for_output(
         &mut self,
-        value: u64,
+        amount: Amount,
         recipient: &PublicAddress,
         memo_context: MemoContext,
     ) -> Result<MemoPayload, NewMemoError> {
         if self.wrote_destination_memo {
             return Err(NewMemoError::OutputsAfterChange);
         }
+        // Check if the outlay is mixing token ids
+        if let Some(prev_token_id) = self.outlay_token_id {
```

Comment:
I wonder if this should be checked in another method? It seems to be mixing the responsibility for `make_memo_for_output` with making sure that the outlay is well-formed?

Reply:
So the idea here is, anything having to do with memos / memo policy is the job of the memo builder.

There is no reason we can't specify RTH to allow some types of mixed transactions -- in fact I think we plan to, and then this wouldn't be an error. But I think it's logically correct to say "today, the RTH scheme is not compatible with mixed transactions, and so it falls to the RTH Memo builder (and not the transaction builder, which is generic over the memo builder) to report that as an error. And then for a future version of RTH memo builder, this may not always be an error.